### PR TITLE
CEDS-1958 Align Caching in Movements & Internal Service

### PR DESCRIPTION
**Pull request by ed0906**
- Removes Keystore from Movements & uses a similar model to the Internal UI
- Uses the JourneyRefiner used in the Internal UI which allows us to have less controller validation

### CODE REVIEW

File: app/controllers/actions/JourneyRefiner.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+  override protected def refine[A](request: AuthenticatedRequest[A]): Future[Either[Result, JourneyRequest[A]]] =
+    refiner(request, Seq.empty[JourneyType]: _*)
+
+  def apply(types: JourneyType*): ActionRefiner[AuthenticatedRequest, JourneyRequest] = new ActionRefiner[AuthenticatedRequest, JourneyRequest] {
+    override protected def refine[A](request: AuthenticatedRequest[A]): Future[Either[Result, JourneyRequest[A]]] =
+      refiner(request, types: _*)
+
+    override protected def executionContext: ExecutionContext = exc
```
Comment (PatrykRudnicki) [inline app/controllers/actions/JourneyRefiner.scala]: hmm, why you need this? You can override thing from implicit. You just need to change name from `exc` to `executionContext`

Comment (PatrykRudnicki) [reply app/controllers/actions/JourneyRefiner.scala]: And line 32 contains this: `override protected def executionContext: ExecutionContext = exc` so you're overriding this two times

Comment (ed0906) [reply app/controllers/actions/JourneyRefiner.scala]: It's an anonymous inner class

Comment (ed0906) [reply app/controllers/actions/JourneyRefiner.scala]: Ah its also a `def` not a val

File: app/connectors/CustomsDeclareExportsMovementsConnector.scala [hunk 2]
```diff
@@ -34,79 +34,58 @@ import scala.util.{Failure, Success}
...
-    postRequest(Departure, movementRequest)
-
-  private def postRequest(
-    actionType: ActionType,
-    movementRequest: MovementRequest
-  )(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse] =
+  def submit(request: MovementRequest)(implicit hc: HeaderCarrier): Future[Unit] =
     httpClient
-      .POST[MovementRequest, HttpResponse](movementSubmissionUrl(actionType), movementRequest, JsonHeaders)
+      .POST[MovementRequest, HttpResponse](appConfig.customsDeclareExportsMovements + Movements, request, JsonHeaders)
```
Comment (bambuchaAdm) [inline app/connectors/CustomsDeclareExportsMovementsConnector.scala]: are we have to do this concatenation for each request ?

File: app/controllers/actions/JourneyRefiner.scala (hunk 1 above)
Comment (bambuchaAdm) [reply app/controllers/actions/JourneyRefiner.scala]: it's dev due to on inline and initialization problems - could be override as val

**Pull request merged**